lib/active_merchant/billing/gateways/payu_latam.rb

```diff
@@ -0,0 +1,484 @@
+require 'digest/md5'
+
+module ActiveMerchant #:nodoc:
+  module Billing #:nodoc:
+    class PayuLatamGateway < Gateway
+      self.display_name = "PayU Latam"
+      self.homepage_url = "http://www.payulatam.com"
+
+      self.test_url = "https://sandbox.api.payulatam.com/payments-api/4.0/service.cgi"
+      self.live_url = "https://api.payulatam.com/payments-api/4.0/service.cgi"
+
+      self.supported_countries = ["AR", "BR", "CO", "MX", "PA", "PE"]
+      self.default_currency = "USD"
+      self.money_format = :dollars
+      self.supported_cardtypes = [:visa, :master, :american_express, :diners_club]
+
+      QUERY_TEST_URL = "https://sandbox.api.payulatam.com/reports-api/4.0/service.cgi"
+      QUERY_LIVE_URL = "https://api.payulatam.com/reports-api/4.0/service.cgi"
+
+      BRAND_MAP = {
+        "visa" => "VISA",
+        "master" => "MASTERCARD",
+        "american_express" => "AMEX",
+        "diners_club" => "DINERS"
+      }
+
+      def initialize(options={})
+        requires!(options, :merchant_id, :account_id, :api_login, :api_key)
+        super
+      end
+
+      def purchase(amount, payment_method, options={})
+        post = {}
+        auth_or_sale(post, 'AUTHORIZATION_AND_CAPTURE', amount, payment_method, options)
+        commit('purchase', post)
+      end
+
+      def authorize(amount, payment_method, options={})
+        post = {}
+        auth_or_sale(post, 'AUTHORIZATION', amount, payment_method, options)
+        commit('auth', post)
+      end
+
+      def capture(authorization, options={})
+        post = {}
+
+        add_credentials(post, 'SUBMIT_TRANSACTION')
+        add_transaction_type(post, 'CAPTURE')
+        add_reference(post, authorization)
+
+        commit('capture', post)
+      end
+
+      def void(authorization, options={})
+        post = {}
+
+        action, _, _ = split_authorization(authorization)
+
+        case action
+        when 'auth'
+          add_credentials(post, 'SUBMIT_TRANSACTION')
+          add_transaction_type(post, 'VOID')
+          add_reference(post, authorization)
+        else
+          message = "Unsupported operation: successful authorize transactions can only be voided."
+          return Response.new(false, message)
+        end
+
+        commit('void', post)
+      end
+
+      def refund(authorization, options={})
+        post = {}
+
+        action, _, _ = split_authorization(authorization)
+
+        case action
+        when 'purchase', 'capture'
+          add_credentials(post, 'SUBMIT_TRANSACTION')
+          add_transaction_type(post, 'REFUND')
+          add_reference(post, authorization)
+        else
+          message = "Unsupported operation: successful purchase and capture transactions can only be refunded."
+          return Response.new(false, message)
+        end
+
+        commit('refund', post)
+      end
+
+      # Supported countries have different minimum amounts and
```

Comment:
I wonder if the currency code factors in.  If they specify the currency in the options, can we lower the amount?

Most gateways have 100 for the `verify` amount.  4000 seems so high!   

I also wonder if for most currencies perhaps 100 might work but for a few currencies we could bump it up to 4000.  Or maybe 4000 all the time is simpler.